game-core/src/main/java/games/strategy/engine/framework/startup/ui/LobbyWatcherKeepAliveTask.java

```diff
@@ -33,7 +34,9 @@
   /** Call this to send a keep-alive request to server. */
   @Nonnull private final Predicate<String> keepAliveSender;
   /** Call this to re-post the current game, obtains a new game id. */
-  @Nonnull private final Supplier<String> gamePoster;
+  @Nonnull private final Supplier<PostGameResponse> gamePoster;
+  /** Call this to shutdown the client in case the server can't establish a connection. */
+  @Nonnull private final Consumer<Boolean> connectivityHandler;
```

Comment:
This should not be needed, if we cannot create an initial connection then we should not be getting into the keep alive tasks (and those should be cancelled already by a failed connectivity check). The overload of keep alive is concerning, doubles the endpoint and I think is a bit of a stretch on SRP.